src/sync_template.sh

```diff
@@ -121,10 +121,33 @@ git commit -m "${PR_COMMIT_MSG}"
 
 echo "::endgroup::"
 
+cleanup_older_prs () {
+  cmd_from_yml_file "precleanup"
+
+  if [ "$IS_DRY_RUN" != "true" ]; then
+    echo "::group::cleanup older PRs"
+    gh pr list \
+    --base "${UPSTREAM_BRANCH}" \
```

Comment:
From a coding point of view this should work.
Just one thing (need to check the details).
I think the `UPSTREAM_BRANCH` is named with a suffix of the commit ID.
Inside the code there will only be a new push/pr if the commit changes.
That means that the `UPSTREAM_BRANCH`  will have a new name if the source repo changes (new commit ID).

Need to check the details (will do today).
An idea is to put a label onto all PRs and search for the label?

Reply:
The UPSTREAM_BRANCH seems to be the one we want to merge into, the one that contains the suffix is NEW_BRANCH if I'm not wrong.

Reply:
Yes indeed, I believe empty labels would cause problems. There are several ways to solve this:
- If PR_LABELS is empty we could skip the cleanup since it would be too dangerous
- Set a default value for PR_LABELS, like `template_sync` or something similar

It's worth noting that if the labels aren't stable between PRs, the cleanup wouldn't work. In my opinion, a small flexibility loss is an acceptable trade off for automation. Of course, this is debatable.

Reply:
hmm I like your suggestion. Maybe let's go for the following:

- if the parameter `pr_labels` is set, we use those labels (like you do currently)
- if the parameter is not set, let's use your suggested label `template_sync`.
- Maybe if that label does not exist, let's create it. Think https://cli.github.com/manual/gh_label_create should be sufficient (not super sure which permissions the action needs https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs), or maybe just add a comment within the documentation?
- document in detail those behaviours

What do you think?

Reply:
That sounds good, I'll make the necessary adjustments.

- Since `pr_labels` will now either be the user-defined list or the default value `template_sync` that list should never be empty unless a user actively decides to use an empty value so I'll guard against that value in the code.
- In the [documentation](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#creating-a-label) it states that anyone with write access to a repository can create labels so that should be fine.
- The user of a default label, and the fact that PR cleanup will be skipped if the label list is empty will be added to the documentation.

Reply:
I've made the necessary changes. I still need to run some test locally to make sure it's all OK but I've pushed the code in case you want to have an early look.

Reply:
Great! and thanks a lot!

I put a tag ([test-prune-pull-requests](https://github.com/AndreasAugustin/actions-template-sync/releases/tag/test-prune-pull-requests)). If you like you can use your version with 

```yaml
        uses: AndreasAugustin/actions-template-sync@test-prune-pull-requests

```

Reply:
I've tested and indeed the PR deletion is not working because I inverted a condition, I will make the code adjustment. Thanks for your help.

Reply:
I created the change locally using a template and a consuming project. It seems to now be working fine using the latest changes.

See below:
<img width="800" alt="Screenshot 2024-01-25 at 11 16 16 AM" src="https://github.com/AndreasAugustin/actions-template-sync/assets/98819045/b95b886a-647b-4d6b-96e7-55b5e169aa9e">